### PR TITLE
content graph - parsing fixes

### PR DESCRIPTION
**Pull request by dantavori**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/content_graph/parsers/integration.py [hunk 1]
```diff
@@ -37,33 +38,34 @@ def connect_to_commands(self) -> None:
...
 
         if default_incident_type := self.yml_data.get('defaultIncidentType'):
             if default_incident_type != 'null':
-                self.add_dependency(default_incident_type, ContentType.INCIDENT_TYPE)
+                self.add_dependency_by_id(default_incident_type, ContentType.INCIDENT_TYPE, is_mandatory=False)
 
     def get_code(self) -> Optional[str]:
         """ Gets the integration code.
-        If the integration is unified, take it from the yml file.
+        If the integration is unified, takes it from the yml file.
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/parsers/integration.py:68]:
```suggestion
        If the integration is unified, then it is taken from the yml file.
```

File: demisto_sdk/commands/content_graph/parsers/script.py [hunk 2]
```diff
@@ -30,18 +30,16 @@ def description(self) -> Optional[str]:
...
+            self.add_command_or_script_dependency(cmd)
 
         for cmd in self.get_command_executions():
-            self.add_dependency(cmd)
+            self.add_command_or_script_dependency(cmd)
 
     def get_code(self) -> Optional[str]:
         """ Gets the script code.
-        If the script is unified, take it from the yml file.
+        If the script is unified, takes it from the yml file.
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/parsers/script.py:42]:
```suggestion
        If the script is unified, it is taken from the yml file.
```

File: demisto_sdk/commands/content_graph/parsers/yaml_content_item.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def deprecated(self) -> bool:
 
     @property
     def description(self) -> Optional[str]:
-        return self.yml_data.get('description')
+        return self.yml_data.get('description', '').replace('\\', '')
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/parsers/yaml_content_item.py:38]: Isn't backslash a newline?
If it is - Need to replace it with a single space, right?

**Review by tkatzir — COMMENTED** [review]
Looks good! Some minor comments and questions.
Let me know when it's ready and I'll approve it.

File: demisto_sdk/commands/content_graph/parsers/yaml_content_item.py (hunk 3 above)
Comment (dantavori) [reply demisto_sdk/commands/content_graph/parsers/yaml_content_item.py:38]: it's allowing a line break in the file but it's not interpreted as a new line.
image
It's causing issues in the CSVs so I removed it. But I'll check if there's a better way to handle this.

Comment (tkatzir) [reply demisto_sdk/commands/content_graph/parsers/yaml_content_item.py:38]: I'd suggest replacing the backslash with a space.
After all the replacements, replace consecutive spaces with a single space.

File: demisto_sdk/commands/content_graph/parsers/yaml_content_item.py [hunk 4]
```diff
@@ -35,7 +35,11 @@ def deprecated(self) -> bool:
 
     @property
     def description(self) -> Optional[str]:
-        return self.yml_data.get('description', '').replace('\\', '')
+        description = self.yml_data.get('description', '')
+        description = description.replace('\\ ', ' ')  # removes unwanted backslashes
+        description = description.replace('\\\n', ' ')  # removes unwanted backslashes
+        description = ' '.join(description.split())  # substitutes multiple spaces into one
+        return description
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/parsers/yaml_content_item.py:42]: Needs a unit test

**Review by tkatzir — APPROVED** [review]
Great! Approved.
If possible, please a unit test (commented.)

**Pull request merged**